### ISSUE TITLE
`esgf-pyclient` automatic download search looks for fx files for a VERY LONG time even if they are not used eventually

### ISSUE THREAD

**Issue by valeriupredoi** [ESMValGroup/ESMValCore#1495]
See @remi-kazeroni [comment](https://github.com/ESMValGroup/ESMValTool/issues/2535#issuecomment-1041823675) - in a nutshell:

- fx files are looked for since they are cell measures attached to some data
- they are probably the worst controlled data on ESGF - some models output them only in one experiment (eg piControl for UKESM1), or not at all, or everywhere for all experiments, and anywhere in between these extremes
- setting the `offline=false` option so that the user can get CMIP data that is truly missing, the auto downloader via esgf-pyclent spends an eternity looking for these, when they may not even be used

Dis a serious performance bug :bug: 

The way I'd fix this would like this: have a look through the preprocessing steps see which fx variable (cell measure) is actually needed, and pass that to datafinder - simple, meerkat :mouse2:

**Comment by valeriupredoi** [comment]
good issue reference @bouweandela :+1: Caching, however nice it is the second time round, does not solve the great wait time first time a recipe is run: just as with buses, if you wait for a bus for an hour, and then two identical buses show up, you're not any less frustrated :bus:

**Comment by remi-kazeroni** [comment]
I'm happy to report that the new version of esgf-pyclient has significantly helped to speed up the search on ESGF nodes 🥳 
A good example is probably `recipe_bock20jgr_fig_8-10`. During the recipe testing for v2.5.0.rc3, the data search on ESGF nodes was not completed after 12 hours. This only took less than 15 minutes for v2.5.0rc4 (using the latest version of esgf-pyclient). Great work @bouweandela! Since I have not seen any problematic case regarding data download in the last round of recipe testing, I think we can close this issue.

**Comment by bouweandela** [comment]
Are you sure you were not using cached search results @remi-kazeroni? I would expect a speedup, but only by a factor of 2 or so.

**Comment by bouweandela** [comment]
Documentation of the caching setting can be found here: https://docs.esmvaltool.org/projects/esmvalcore/en/latest/quickstart/configure.html#search

**Comment by remi-kazeroni** [comment]
> Are you sure you were not using cached search results @remi-kazeroni? I would expect a speedup, but only by a factor of 2 or so.

Yes I used cached results with the same settings for the past three rounds of recipe testing including for v2.5.0rc4. The tool was looking indefinitely for fx files before and for some reason that went much faster yesterday. I also did not run into any ESGF connection problem which was the case before. Perhaps a larger number of nodes were available yesterday than for the previous rounds and that helped running recipes.
